src/frontend/overlay/src/components/atoms/ContestLabels.jsx

```diff
@@ -115,27 +117,52 @@ const TaskResultLabelWrapper2 = styled.div`
   color: #fff;
 `;
 
-// TODO: fts start
-const ICPCTaskResultLabel2 = ({ problemResult: r, ...props }) => {
+
+const StarIconWrap = styled.div`
+    position: absolute;
+    width: 33px;
+    height: 33px;
+    mask: url(${star});
+    background-position: center;
+    background-color: ${({ color }) => color};
+`;
+
+const StarIcon = ({ color }) => {
+    return <StarIconWrap color={color}/>;
+};
+
+const AttemptsOrScoreLabelWrapper = styled.div`
+    position: relative;
+`;
+
+const defaultColorForStar = "#000";
+const greenColor = "#3bba6b";
+
+const ICPCTaskResultLabel2 = ({ problemColor, problemResult: r, ...props }) => {
     const status = getStatus(r.isFirstToSolve, r.isSolved, r.pendingAttempts, r.wrongAttempts);
     const attempts = r.wrongAttempts + r.pendingAttempts;
     return <>
-        {/*{status === TeamTaskStatus.first && <StarIcon/>}*/}
         <TaskResultLabelWrapper2 color={TeamTaskColor[status]} {...props}>
-            {TeamTaskSymbol[status]}
-            {status !== TeamTaskStatus.untouched && attempts > 0 && attempts}
+            { status === TeamTaskStatus.first && <StarIcon color={(problemColor === undefined || problemColor === greenColor) ? defaultColorForStar : problemColor}/> }
+            <AttemptsOrScoreLabelWrapper>
+                {TeamTaskSymbol[status]}
+                {status !== TeamTaskStatus.untouched && attempts > 0 && attempts}
+            </AttemptsOrScoreLabelWrapper>
         </TaskResultLabelWrapper2>
     </>;
 };
 
-const IOITaskResultLabel2 = ({ problemResult: r, minScore, maxScore,  ...props }) => {
+const IOITaskResultLabel2 = ({ problemColor, problemResult: r, minScore, maxScore,  ...props }) => {
```

Comment:
Is it really necessary to rename problemResult to r?

Reply:
I didn't to it. That was before I started pr.

Reply:
Then I guess we will refactor it some time in the future.